### PR TITLE
Run2-hcx168 Add tests for depth collapsing in HCAL reconstruction

### PR DESCRIPTION
**Pull request by bsunanda**
Needs new geometry for 2018 HCAL and a change in the era for Run_2018 to do the collapsing of HE depths during local reconstruction

### CODE REVIEW

File: RecoLocalCalo/HcalRecAlgos/test/HcalCollapseAnalyzer.cc [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+void HcalCollapseAnalyzer::beginJob() {
+  // Book histograms
+  edm::Service<TFileService>             fs;
+  h_merge  = fs->make<TH1I>("h_merge","Number of hits merged",10,0.0,10.0);
+  h_size   = fs->make<TH1I>("h_size","Size of the RecHit collection",100,500.0,1500.0);
+  h_depth  = fs->make<TH1I>("h_depth","Depth of the Id's used",10,0.0,10.0);
+  h_sfrac  = fs->make<TH1D>("h_sfrac","Ratio of sizes of preRecHit and RecHit collections",150,0.0,1.5);
+  h_frac   = fs->make<TH1D>("h_frac", "Fraction of energy before collapse",150,0.0,1.5);
+  h_balance= fs->make<TH1D>("h_balance", "Balance of energy between pre- and post-collapse",100,0.5,1.5);
```
Comment (slava77) [inline RecoLocalCalo/HcalRecAlgos/test/HcalCollapseAnalyzer.cc]: This module only fills per-event histograms. This looks like a good case to make it a DQMEDAnalyzer and perhaps even move it to a relevant DQM package.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: RecoLocalCalo/HcalRecAlgos/test/HcalCollapseAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply RecoLocalCalo/HcalRecAlgos/test/HcalCollapseAnalyzer.cc]: Did not realize it will be reviewed so quickly. OK could be moved

**Comment by davidlange6** [comment]
please test

**Comment by bsunanda** [comment]
@dmitrijus Can you please approve this

**Comment by dmitrijus** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**